crates/sui-source-validation/src/lib.rs

```diff
@@ -617,15 +617,37 @@ fn download_and_compile(
         // Check the platform and proceed if we can download a binary. If not, the user should follow error instructions to sideload the binary.
         let platform = detect_platform(&root, compiler_version, &dest_canonical_binary)?;
         // Download if binary does not exist.
-        let url = format!("https://github.com/MystenLabs/sui/releases/download/mainnet-v{}/sui-mainnet-v{}-{}.tgz", compiler_version, compiler_version, platform);
+        let mainnet_url = format!(
+            "https://github.com/MystenLabs/sui/releases/download/mainnet-v{}/sui-mainnet-v{}-{}.tgz",
+            compiler_version, compiler_version, platform
+        );
 
         println!(
-            "{} compiler @ {} (this may take a while)",
+            "{} mainnet compiler @ {} (this may take a while)",
             "DOWNLOADING".bold().green(),
             compiler_version.yellow()
         );
 
-        let mut response = ureq::get(&url).call()?.into_reader();
+        let mut response = match ureq::get(&mainnet_url).call() {
```

Comment:
You might be able to write this more cleanly in its own function, then you can adopt a pattern of:

```rust
match get(mainnet_url) {
    Ok(response) => return Ok(response);
    Err(Status(404, _)) => { /* fallthrough */ }
    Err(e) => return Err(e.into());
};

get(testnet_url).map_err(|_|...)
```

Reply:
Hmmm I can't get that to typecheck, maybe I'm missing something. Seems `{ /* fallthrough */ }` has to return a `Result` type